src/shared/api/firebase/auth.ts

```diff
@@ -0,0 +1,30 @@
+import { auth } from '@/shared/config/firebase';
+import {
+  createUserWithEmailAndPassword,
+  signInWithPopup,
+  GoogleAuthProvider,
+  type UserCredential,
+  signInWithEmailAndPassword,
+  signOut,
+} from 'firebase/auth';
+
+export type SignUpPayload = { email: string; password: string };
+export type SignInPayload = { email: string; password: string };
```

Comment:
Can we give it a single common name and use it as one type?
For example

```suggestion
export type AuthPayload = { email: string; password: string };
```